Library/ViewModels/ShippingRulesViewModel.swift

```diff
@@ -32,70 +33,73 @@ public protocol ShippingRulesViewModelType {
 
-    let reloadDataWithShippingRulesSelected = dataSelected
-      .map(shippingRuleData(for:shippingRules:selectedShippingRule:))
-      .map { ($0, false) }
+    let filteredData = initialData
+      .takePairWhen(searchText)
+      .map(dataMatching(_:searchText:))
 
-    self.reloadDataWithShippingRules = Signal.merge(
-      reloadDataWithShippingRulesInitial,
-      reloadDataWithShippingRulesSelected
+    let selectedShippingRule = Signal.merge(
+      initialData,
+      filteredData
     )
+    .takePairWhen(selectedIndex)
+    .map { data, selectedIndex in (data.1, selectedIndex) }
+    .map { shippingRules, selectedIndex in shippingRules[selectedIndex] }
 
-    self.flashScrollIndicators = reloadDataWithShippingRulesInitial
-      .ignoreValues()
-
-    let selectedShippingRuleIndexInitial = dataInitial
-      .map { _, shippingRules, selectedShippingRule in shippingRules.firstIndex(of: selectedShippingRule) }
-      .skipNil()
+    self.notifyDelegateOfSelectedShippingRule = selectedShippingRule
+      .skipRepeats()
 
-    let selectedShippingRuleIndexSelected = self.didSelectShippingRuleAtIndexProperty.signal
-      .skipNil()
+    let reloadDataInitial = initialData
+      .map(shippingRulesData(project:shippingRules:selectedShippingRule:))
+      .map { ($0, true) }
 
-    self.notifyDelegateOfSelectedShippingRule = dataSelected
-      .map(third)
+    let reloadDataFiltered = filteredData
+      .withLatest(from: Signal.merge(initialData.map(third), selectedShippingRule))
+      .map { data, selectedShippingRule in (data.0, data.1, selectedShippingRule) }
+      .map(shippingRulesData(project:shippingRules:selectedShippingRule:))
+      .map { ($0, true) }
 
-    self.scrollToCellAtIndex = Signal.combineLatest(
-      selectedShippingRuleIndexInitial,
-      reloadDataWithShippingRulesInitial
+    let reloadDataSelected = Signal.merge(
+      initialData,
+      filteredData
     )
-    .map(first)
+    .takePairWhen(selectedIndex)
+    .map { data, selectedIndex in (data.0, data.1, selectedIndex) }
+    .map { project, shippingRules, selectedIndex in (project, shippingRules, shippingRules[selectedIndex]) }
+    .map(shippingRulesData(project:shippingRules:selectedShippingRule:))
+    .map { ($0, false) }
+
+    self.flashScrollIndicators = initialData
+      .take(first: 1)
+      .ignoreValues()
 
-    self.selectCellAtIndex = Signal.merge(
-      selectedShippingRuleIndexInitial,
-      selectedShippingRuleIndexSelected
+    self.reloadDataWithShippingRules = Signal.merge(
+      reloadDataInitial,
+      reloadDataFiltered,
+      reloadDataSelected
     )
-    .skipRepeats()
 
-    self.deselectCellAtIndex = self.selectCellAtIndex
-      .combinePrevious()
-      .map(first)
+    self.scrollToCellAtIndex = initialData
+      .take(first: 1)
+      .map { _, shippingRules, selectedShippingRule in shippingRules.firstIndex(of: selectedShippingRule) }
+      .skipNil()
+
+    self.selectCellAtIndex = self.deselectCellAtIndex
```

Comment:
This is very confusing. Can't it just be `self.selectedCellAtIndex = selectedIndex`?